Lib/test/test_os.py

```diff
@@ -835,6 +835,21 @@ def test_key_type(self):
         self.assertIs(cm.exception.args[0], missing)
         self.assertTrue(cm.exception.__suppress_context__)
 
+    def test_iter_error_when_os_environ_changes(self):
+        def _iter_environ_change():
```

Comment:
You can just use `iter(os.environ.items())`.

It may be worth to test also other iterators: `iter(os.environ)` and `iter(os.environ.values())`.